encord/objects/coordinates.py

```diff
@@ -150,29 +152,92 @@ class Visibility(Flag):
     OCCLUDED = auto()
 
 
-@dataclass(frozen=True)
+@dataclass(init=False)
 class SkeletonCoordinate:
     x: float
     y: float
 
     # `name` and `color` can be removed as they are part of the ontology.
     # The frontend must first be aware of how to merge with ontology.
     name: str
-    color: str
+    color: Optional[str] = "#00000"
 
     # `featureHash` and `value` seem to appear when visibility is
     # present. They might not have any meaning. Remove if confirmed that
     # Frontend does not need it.
-    feature_hash: Optional[str] = None
+    featureHash: Optional[str] = None
     value: Optional[str] = None
 
-    visibility: Optional[Visibility] = None
+    def __init__(
+        self,
+        x: float,
+        y: float,
+        name: str,
+        color: str = "#00000",
+        featureHash: Optional[str] = None,
+        value: Optional[str] = None,
+    ) -> None:
+        self.x = x
+        self.y = y
+        self.name = name
+        self.color = color
+        self.featureHash = featureHash if featureHash else ""
+        self.value = value if value else name
+
+    @staticmethod
+    def from_dict(d: dict[str, Any]) -> SkeletonCoordinate:
+        return SkeletonCoordinate(**d)
+
+    def to_dict(self) -> dict:
+        return {
+            "x": self.x,
+            "y": self.y,
+            "name": self.name,
+            "color": self.color,
+            "value": self.value,
+            "featureHash": self.featureHash,
+        }
 
 
 @dataclass(frozen=True)
 class SkeletonCoordinates:
     values: List[SkeletonCoordinate]
 
+    @staticmethod
+    def from_dict(d: dict) -> SkeletonCoordinates:
+        skeleton_dict = d["skeleton"]
+        list_coordinates: List[SkeletonCoordinate] = []
+        sorted_dict_value_tuples = sorted((int(key), value) for key, value in skeleton_dict.items())
+        sorted_dict_values = [item[1] for item in sorted_dict_value_tuples]
+        for coordinate in sorted_dict_values:
+            skeleton_coordinate = SkeletonCoordinate(
+                x=coordinate["x"],
+                y=coordinate["y"],
+                name=coordinate["name"],
+                value=coordinate["value"],
+                color=coordinate["color"],
+                featureHash=coordinate["featureHash"],
+            )
+            list_coordinates.append(skeleton_coordinate)
+        return SkeletonCoordinates(values=list_coordinates)
+
+    def to_dict(self) -> dict:
+        return {i: x.to_dict() for i, x in enumerate(self.values)}
+
+
+# @dataclass
```

Comment:
Aim would be in time that SkeletonTemplate.create_instance(list[SkeletonCoordinate]) -> SkeletonInstance, so we move away from SkeletonCoordinates.